flatsurf/geometry/hyperbolic.py

```diff
@@ -14266,3 +14408,34 @@ def _lt_(cls, lhs, rhs):
         return geodesic.parametrize(
             lhs.start(), model="euclidean"
         ) < geodesic.parametrize(rhs.start(), model="euclidean")
+
+
+def boundary_point_is_between(p0, p1, q):
```

Comment:
Is this not just the question whether q is in the half space left or right of the geodesic p0-p1?

Reply:
It is!